[IMP] payroll: update work entries, add menu

larm-odoo opened the pull request:
This doc updates the work entries section of the Payroll app. There is a new section, Time Off To Report, that is new for 17.0.

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/documentation/pull/7551).

jero-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -80,95 +93,186 @@ there are any errors on the work entry, such as required fields being left blank
+the left side of the pop-up. The :guilabel:`From` and :guilabel:`To` date and time range, as well as
+the total time (in hours), appears in the :guilabel:`Duration` field on the right side.
+
+If the conflict is due to a time-off request that has not been approved yet, a :guilabel:`Time Off`
```
```suggestion
If the conflict is due to a time off request that has not been approved yet, a :guilabel:`Time Off`
```

jero-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -80,95 +93,186 @@ there are any errors on the work entry, such as required fields being left blank
+
+The most common work entry conflicts are for time off requests that have been submitted but not yet
+approved, which result in duplicate work entries for that employee (one for time-off and another for
+regular work). If there is a conflict because a time-off request is in the system for the same time
+that a regular work entry already exists, the time-off request is entered in the :guilabel:`Time
```
```suggestion
regular work). If there is a conflict because a time off request is in the system for the same time
that a regular work entry already exists, the time off request is entered in the :guilabel:`Time
```

jero-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -80,95 +93,186 @@ there are any errors on the work entry, such as required fields being left blank
-If there is a conflict because a time-off request for the same time already exists in the system,
-the time-off will be entered in the :guilabel:`Time Off` field. Clicking the :guilabel:`External
-Link` button next to the :guilabel:`Time Off` entry will bring up the duplicate time-off request.
+The time-off conflict can be resolved either on the work entry pop-up, or on a detailed time-off
+request pop-up. To resolve the time-off conflict on this work entry pop-up, click the
+:guilabel:`Approve Time Off` button to approve the time off request and resolve the work entry
+conflict.
```
```suggestion
The time off conflict can be resolved either on the work entry pop-up, or on a detailed time off
request pop-up. To resolve the time off conflict on this work entry pop-up, click the
:guilabel:`Approve Time Off` button to approve the time off request and resolve the work entry
conflict.
```

jero-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -80,95 +93,186 @@ there are any errors on the work entry, such as required fields being left blank
+.. image:: work_entries/approve-work-entry.png
    :align: center
-   :alt: Edit and/or validate a duplicate time-off request.
+   :alt: The time-off conflict in the work entry detail pop-up.
```
```suggestion
   :alt: The time off conflict in the work entry detail pop-up.
```

jero-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -80,95 +93,186 @@ there are any errors on the work entry, such as required fields being left blank
-After conflicts have been resolved, work entries must be regenerated for each employee by clicking
-the :guilabel:`Regenerate Work Entries` button, and entering the corresponding information for each
-employee.
+To resolve the time-off conflict on the detailed time-off request pop-up, click the
+:guilabel:`Internal Link` button at the end of the :guilabel:`Time Off` entry line and the time-off
+request details appear in a pop-up window. The request can be modified if needed.
```
```suggestion
To resolve the time-off conflict on the detailed time off request pop-up, click the
:guilabel:`Internal Link` button at the end of the :guilabel:`Time Off` entry line and the time off
request details appear in a pop-up window. The request can be modified if needed.
```

jero-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -80,95 +93,186 @@ there are any errors on the work entry, such as required fields being left blank
+.. image:: work_entries/time-off-details.png
    :align: center
-   :alt: Regenerate Work Entries button on the Work Entries Regeneration form.
+   :alt: The detailed time-off request form.
```
```suggestion
   :alt: The detailed time off request form.
```

jero-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -80,95 +93,186 @@ there are any errors on the work entry, such as required fields being left blank
+is visible. If the approval was a mistake, the request can be refused here by clicking the
+:guilabel:`Refuse Time Off` button.
+
+Since the time-off was approved in the time-off window, click the :guilabel:`X` in the top right
+corner to close the window. The conflict disappears from the work entry dashboard since the conflict
+is resolved.
```
```suggestion
Since the time off was approved in the time off window, click the :guilabel:`X` in the top right
corner to close the window. The conflict disappears from the work entry dashboard since it has been resolved.
```

jero-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -80,95 +93,186 @@ there are any errors on the work entry, such as required fields being left blank
+Time Off to Report
+==================
+
+If a time-off request is submitted for a time period that was already processed on a payslip, the
+time-off request appears in the :guilabel:`Time Off to Report`, with a status of :guilabel:`To defer
+to next payslip`. This is because the employee was already paid for that day, and it was logged as
+time spent at work, as a typical work day.
```
```suggestion
If a time off request is submitted for a time period that was already processed on a payslip, the
time off request appears in the :guilabel:`Time Off to Report`, with a status of :guilabel:`To defer
to next payslip`. This is because the employee was already paid for that day, and it was logged as
time spent at work, as a typical work day.
```

jero-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -80,95 +93,186 @@ there are any errors on the work entry, such as required fields being left blank
+to next payslip`. This is because the employee was already paid for that day, and it was logged as
+time spent at work, as a typical work day.
+
+In order to keep the employee's time-off balances correct, the time-off request must be applied to
+the following pay period. This not only ensures time-off request balances are current, but this also
+eliminates the need to redo work entries, cancel paychecks, and reissue paychecks.
```
```suggestion
In order to keep the employee's time off balances correct, the time off request must be applied to
the following pay period. This not only ensures time off request balances are current, but it also
eliminates the need to redo work entries, cancel paychecks, and reissue paychecks.
```

jero-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -80,95 +93,186 @@ there are any errors on the work entry, such as required fields being left blank
+
+The most common scenario when this situation occurs, is when payslips are processed a day or two
+before the pay period ends, and an employee is unexpectedly sick on one of the last days of the pay
+period. The employee puts in a time-off request for a day that was already processed on a payslip as
+a regular work day. Instead of canceling the payslip, modifying the work entries, and reissuing the
+paycheck, Odoo allows for those time-off requests to be applied to the following pay period instead.
```
```suggestion
period. The employee puts in a time off request for a day that was already processed on a payslip as
a regular work day. Instead of canceling the payslip, modifying the work entries, and reissuing the
paycheck, Odoo allows for those time off requests to be applied to the following pay period instead.
```

jero-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -80,95 +93,186 @@ there are any errors on the work entry, such as required fields being left blank
+a regular work day. Instead of canceling the payslip, modifying the work entries, and reissuing the
+paycheck, Odoo allows for those time-off requests to be applied to the following pay period instead.
+
+To view all the time-off requests that need to be deferred to the next payslip, navigate to
```
```suggestion
To view all the time off requests that need to be deferred to the next payslip, navigate to
```

jero-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -80,95 +93,186 @@ there are any errors on the work entry, such as required fields being left blank
+:menuselection:`Payroll app --> Work Entries --> Time Off to Report`. The default filter for this
+report is :guilabel:`To Defer`.
+
+All time-off requests that need to be applied to the following pay period appear, with a
```
```suggestion
All time off requests that need to be applied to the following pay period appear with a
```

jero-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -80,95 +93,186 @@ there are any errors on the work entry, such as required fields being left blank
+
+.. image:: work_entries/time-off-to-report.png
+   :align: center
+   :alt: A list of all time-off requests that were not approved before payslips were generated.
```
```suggestion
   :alt: A list of all time off requests that were not approved before payslips were generated.
```

jero-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -80,95 +93,186 @@ there are any errors on the work entry, such as required fields being left blank
+   :align: center
+   :alt: A list of all time-off requests that were not approved before payslips were generated.
+
+Defer multiple time-off requests
```
```suggestion
Defer multiple time off requests
```

jero-odoo replied:
Should this heading be "defer multiple work entries"? Time off requests are not mentioned in this section, only work entries.

larm-odoo replied:
I am torn because the only work entries that can be deferred are ones related to time off (like vacation, sick time, etc). So even though they are work entries, it is only for specific work entries that are for a time off type. A regular work entry would not appear here. In the settings section, the option to "Defer Time Off" appears in the "Time Off" section since it only applies to time off entries. SO I thought to be very clear, I would say time off in this heading so users know that it's only for time off entries.

jero-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -80,95 +93,186 @@ there are any errors on the work entry, such as required fields being left blank
+   :align: center
+   :alt: The actions button and # Selected buttons that appear after any selections are made.
+
+Defer individual time-off requests
+----------------------------------
+
+Time-off requests appearing on the :guilabel:`Time Off to Report` list can be deferred individually.
+
+Click on an individual time-off request and the details for that request load. The specific details
+for the time-off request appear on the left-hand side, and all of the employee's submitted time-off
+requests appear on the right-hand side (including the request in the details on the left-hand side).
```
```suggestion
Defer individual time off requests
----------------------------------

Time-off requests appearing on the :guilabel:`Time Off to Report` list can be deferred individually.

Click on an individual time off request and the details for that request load. The specific details
for the time off request appear on the left-hand side, and all of the employee's submitted time off
requests appear on the right-hand side (including the request in the details on the left-hand side).
```

jero-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -80,95 +93,186 @@ there are any errors on the work entry, such as required fields being left blank
+for the time-off request appear on the left-hand side, and all of the employee's submitted time-off
+requests appear on the right-hand side (including the request in the details on the left-hand side).
+
+To defer the time-off request to the next payslip, click the :guilabel:`Report to Next Month` button
+at the top. Once processed, the :guilabel:`Report to Next Month` button disappears, and the
+:guilabel:`Payslip State` changes from :guilabel:`To defer to next payslip` to :guilabel:`Computed
+in Current Payslip`.
```
```suggestion
To defer the time off request to the next payslip, click the :guilabel:`Report to Next Month` button
at the top. Once processed, the :guilabel:`Report to Next Month` button disappears, and the
:guilabel:`Payslip State` changes from :guilabel:`To defer to next payslip` to :guilabel:`Computed
in Current Payslip`.
```

jero-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -80,95 +93,186 @@ there are any errors on the work entry, such as required fields being left blank
+
+.. image:: work_entries/single-defer.png
+   :align: center
+   :alt: The time-off details for an individual request that needs to be deferred.
```
```suggestion
   :alt: The time off details for an individual request that needs to be deferred.
```

jero-odoo reviewed (changes requested):
hey @larm-odoo let me know if you have any questions. I noticed there were a lot of instances of "time-off" used instead of "time off". I tried to see if this was one of the times where we match the UI, but I couldn't see any use of it written that way in Odoo or in your other docs, so I switched them to all be "time off".

larm-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -2,71 +2,84 @@
+alphabetical order, by employee first name. The entire current months is displayed, with the current
+day highlighted in pale yellow.
+
+The default filter enabled on the *Work Entries* dashboard is `Conflicting`, which displays only
+entries with a conflict that need to be resolved.
```
I love that suggestion! Done!

larm-odoo commented:
Thanks @jero-odoo - I made all the updates and commented on the section titles. I also found a couple of other places I had future tense ("sill") so I also made those changes. I don't know why I assumed it was "Time-off"- that was not on my radar for whatever reason! So those were all changed.

jero-odoo commented:
@larm-odoo per our conversation, I am holding off on reviewing this one for now, let me know when you are ready for me to take a look at it 👍

larm-odoo commented:
Hi @jero-odoo this is ready, again. I saw what happened. I had the new linter installed, and I had a few different tabs/docs open because I was looking at different docs I wrote for reference on a few things. When I saw the 'warnings' in the linter, I made edits to the payslips doc and not the work entries doc! So now I reverted the payslips doc so it doesn't appear as changes on this doc. What a journey!

jero-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -80,95 +96,186 @@ there are any errors on the work entry, such as required fields being left blank
-the left side of the pop-up. The :guilabel:`From` and :guilabel:`To` date range, as well as the
-total time requested (in hours), appear in the :guilabel:`Period` field on the right side.
+The most common work entry conflicts are for time off requests that have been submitted but not yet
+approved, which results in duplicate work entries for that employee (one for time off and another for
```
Check line length (over 100 characters)

jero-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -80,95 +96,186 @@ there are any errors on the work entry, such as required fields being left blank
 
-Conflicts are indicated with an orange triangle in the top left corner of each individual entry.
-Click on an individual work entry to see the conflict details in a pop-up.
+Conflicts are indicated with an orange triangle in the top left corner of each individual work
```
```suggestion
Conflicts are indicated with an orange triangle in the top-left corner of each individual work
```

jero-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -80,95 +96,186 @@ there are any errors on the work entry, such as required fields being left blank
+is visible. If the approval was a mistake, the request can be refused here by clicking the
+:guilabel:`Refuse Time Off` button.
+
+Since the time off was approved in the time off window, click the :guilabel:`X` in the top right
```
```suggestion
Since the time off was approved in the time off window, click the :guilabel:`X` in the top-right
```

jero-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -80,95 +96,186 @@ there are any errors on the work entry, such as required fields being left blank
 Click the :guilabel:`Create Draft Entry` button to create the payslips for the batch.
 
-Click on the :guilabel:`Payslips` button in the top right to view all the payslips for the batch.
+Click the :guilabel:`Payslips` smart button in the top right to view all the payslips for the batch.
```
```suggestion
Click the :guilabel:`Payslips` smart button in the top-right to view all the payslips for the batch.
```

jero-odoo reviewed (approved):
Hey @larm-odoo just a few small formatting changes, but otherwise this one is ready for the next round.

StraubCreative reviewed (approved):
hi @larm-odoo 
I have a couple things for you from 2nd-round PR.
Once these are addressed it should be ready for the next stage.
Thanks!

larm-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -80,95 +96,186 @@ there are any errors on the work entry, such as required fields being left blank
+Defer individual time off entries
+---------------------------------
+
+Time-off requests appearing on the :guilabel:`Time Off to Report` list can be deferred individually.
```
Thanks Zac- I checked for "time-off" (lowercase) in  a previous edit, so it didn't catch this one - I forgot VSCode seems to be case-sensitive. All changed now!

larm-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -2,76 +2,92 @@
-After a work entry has been either added or modified, the work entries need to be regenerated for
-the affected employee(s). Click on the :guilabel:`Regenerate Work Entries` button at the top of the
-main dashboard, and a pop-up will appear.
+After work entries have been either added or modified, the work entries need to be regenerated for
+the affected employees. Click on the :guilabel:`Regenerate Work Entries` button at the top of the
```
Lol- good point! First, I agree with using "they" - I just wanted to be 110% clear, which is why I used "work entries" again. Second-  I had checked with the HR experts, and they need to be regenerated when they have been changed or added, because (I think, but will double check) they were already generated once. When they were modified or added after the fact, they must be regenerated so they are computed correctly for the payslips.

So if I have a work entry that said I was working 8 hours that day but I actually only worked 4 (maybe because of sick time, vacation time, or just a regular old no-show) then the entry has to be updated to reflect what happened. Since they were already 'generated' they need to be 'regenerated' so they are updated and correct before issuing payslips.

I will re-check with the experts to make sure I am understanding this correctly, and I will expand on this so it is understood why we do this. Ill be honest, I have a 90% grasp on this but the last 10% is still "magic" to me =D

larm-odoo replied:
OK, I was completely WRONG on how regenerating work entries works!! I chatted with Emilio (who has been helping me with HR questions) and he explained this for me. I have now updated this section, moved it (it didn't make sense to keep it here) and I also added a sentence in the very beginning of the doc, so this section makes sense. 

TL/DR: If conflicts happen form an app (time off, planning, etc) you can/should resolve the conflict in the APP that created the issue. THEN, you regenerate the work entries in payroll, which deletes the old entries and pulls new ones for the respective apps.

This keeps the data across ALL apps correct instead of just fixing them in the payroll app. Because if you DON'T fix it, then if you regenerate work entries- the issue appears again.

larm-odoo commented:
Hi @StraubCreative - I fixed the issues you found, plus reworked a section (and moved it) because I was WRONG with how regenerating work entries worked! I rechecked with the HR people (I had originally asked them when writing my first work entries doc, and clearly got it wrong). So that has been corrected- and I had to add a sentence to the beginning to make it all come together.  I also realized I tagged you because when you let me know about my conflict/error in the build, for some reason I thought you had reviewed the doc (incorrect). So I apologize for tagging you sooner- I just clicked the 're-request' review button. 

OK, on to final review!

ksc-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -2,173 +2,304 @@
+Add a new work entry
+====================
+
+If a work entry is missing and needs to be added, such as sick time, or if an employee forgot to
+clock in and out for a shift, click :guilabel:`New` to create a new work entry. A :guilabel:`Create`
+work entry pop-up form appears. Enter the following information on the form:
+
+- :guilabel:`Description`: enter a short description for the work entry, such as `Sick Time`. If
+  this field is left blank, it automatically populates once an employee is selected. The default
+  entry is `Attendance: (Employee)`.
+- :guilabel:`Employee`: select the employee the work entry is for, using the drop-down menu.
+- :guilabel:`Work Entry Type`: select the :ref:`work entry type <payroll/work-entries-config>` using
+  the drop-down menu.
+- :guilabel:`From` and :guilabel:`To`: enter the start (:guilabel:`From`) and end (:guilabel:`To`)
+  dates and times for the work entry. First, click on either the :guilabel:`From` or :guilabel:`To`
+  line and a calendar pop-up appears. Select the date by navigating to the correct month and year
+  using the :guilabel:`< (left arrow)` and :guilabel:`> (right arrow)` icons, then click on the
+  specific day.
+
+  Next, select the time by clicking on either the hour or minute fields at the bottom of the
+  calendar, and select the desired time for both the hour and minutes.
+
+  When the date and time are correct for the entry, click the :guilabel:`Apply` button.
+- :guilabel:`Duration`: displays the hours based on the :guilabel:`To` and :guilabel:`From` entries.
+  Modifying this field modifies the :guilabel:`To` field (the :guilabel:`From` field does not
+  change).
```
```suggestion
If a work entry is missing and needs to be added, such as sick time, or if an employee forgot to
clock in and out for a shift, click :guilabel:`New` on the :guilabel:`Work Entry` dashboard, to
create a new work entry.

A :guilabel:`Create` work entry pop-up form appears.

Enter the following information on the form:

- :guilabel:`Description`: enter a short description for the work entry, such as `Sick Time`. If
  this field is left blank, it automatically populates once an employee is selected. The default
  entry is `Attendance: (Employee)`.
- :guilabel:`Employee`: select the employee the work entry is for, using the drop-down menu.
- :guilabel:`Work Entry Type`: select the :ref:`work entry type <payroll/work-entries-config>` using
  the drop-down menu.
- :guilabel:`From` and :guilabel:`To`: enter the start (:guilabel:`From`) and end (:guilabel:`To`)
  dates and times for the work entry.
  
  First, click on either the :guilabel:`From` or :guilabel:`To` line to reveal a calendar pop-up
  window. Select the date by navigating to the correct month and year, using the :guilabel:`< (left
  arrow)` and :guilabel:`> (right arrow)` icons, then click on the specific day.

  Next, select the time, by clicking on either the hour or minute fields at the bottom of the
  calendar, and select the desired time for both the hour and minutes.

  When the date and time are correct for the entry, click the :guilabel:`Apply` button.
- :guilabel:`Duration`: displays the hours based on the :guilabel:`To` and :guilabel:`From` entries.
  Modifying this field modifies the :guilabel:`To` field (the :guilabel:`From` field does not
  change).
```

ksc-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -2,173 +2,304 @@
-   Before clicking either :guilabel:`Save & Close` or :guilabel:`Save & New`, it is good to double
-   check the :guilabel:`Period` to ensure the time stated corresponds to the :guilabel:`To` and
-   :guilabel:`From` fields.
+Any work entry that has a conflict to be resolved is indicated on the main :guilabel:`Work Entry`
+dashboard. Navigate to :menuselection:`Payroll app --> Work Entries --> Work Entries`. Only
+conflicts needing resolution are shown by default.
```
```suggestion
Any work entry that has a conflict to be resolved is indicated on the main :guilabel:`Work Entry`
dashboard, which can be accessed by navigating to :menuselection:`Payroll app --> Work Entries -->
Work Entries`. Only conflicts needing resolution are shown by default.
```

ksc-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -2,173 +2,304 @@
 
-Once the information is entered, click :guilabel:`Save & Close` to save the entry and close the
-pop-up, or :guilabel:`Save & New` to save the entry and create another :guilabel:`Work Entry Type`.
+The conflict is briefly explained in an orange text box.
```
I also the doc could benefit from a screenshot of this pop-up, as well

larm-odoo replied:
I added it- then realized the one directly beneath it was the same image with different call-outs. So I deleted the subsequent one and kept the new one here.

ksc-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -2,173 +2,304 @@
-Click the :guilabel:`Regenerate Work Entries` button and the work entries will be recreated. Once
-finished, the pop-up will close.
+
+Click the :guilabel:`Regenerate Work Entries` button and the work entries are recreated. Once
+finished, the pop-up closes.
```
```suggestion
Click the :guilabel:`Regenerate Work Entries` button, and the work entries are recreated. Once
finished, the pop-up window closes.
```

ksc-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -2,173 +2,304 @@
-A conflict appears for any request that has not been approved, such as sick time or vacation, or if
-there are any errors on the work entry, such as required fields being left blank. Conflicts are
-required to be resolved before payslips can be generated.
+   To correct this issue, modify the employee's schedule in the *Planning* app so they are correctly
+   assigned to only one work station. Then, in the *Payroll* app, regenerate work entries for that
+   employee, for that time period.
```
```suggestion
   To correct this issue, modify the employee's schedule in the *Planning* app so, they are correctly
   assigned to only one work station. Then, in the *Payroll* app, regenerate work entries for that
   employee, for that specific time period.
```

ksc-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -2,173 +2,304 @@
 
-The conflict is briefly explained in an orange text box.
+.. tip::
+   If the :guilabel:`Generate Payslips` button is not active (appears pale purple instead of dark
+   purple), that indicates there are conflicts, or the date selected includes dates in the future.
+   Resolve all conflicts before generating payslips.
```
```suggestion
   If the :guilabel:`Generate Payslips` button is not active (appears pale purple, instead of dark
   purple), that indicates there are conflicts, or the date selected includes dates in the future.
   Resolve all conflicts before generating payslips.
```

ksc-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -2,173 +2,304 @@
-The :guilabel:`Work Entry Name`, :guilabel:`Employee`, and :guilabel:`Work Entry Type` are listed on
-the left side of the pop-up. The :guilabel:`From` and :guilabel:`To` date range, as well as the
-total time requested (in hours), appear in the :guilabel:`Period` field on the right side.
+The date range the payslips apply to appears in the :guilabel:`Period` field, and the company
+appears in the :guilabel:`Company` field. It is not possible to make changes to this form.
```
```suggestion
The date range to which the payslips apply appears in the :guilabel:`Period` field, and the company
appears in the :guilabel:`Company` field. It is **not** possible to make changes to this form.
```

ksc-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -2,173 +2,304 @@
-The details for the time-off request appear in the pop-up. The request can be modified if needed.
-Click either the :guilabel:`Validate` or :guilabel:`Refuse` button to approve or deny the request,
-then click the :guilabel:`Save` button to save the changes.
+Click the :guilabel:`Payslips` smart button in the top-right to view all the payslips for the batch.
```
```suggestion
Click the :guilabel:`Payslips` smart button at the top of the page to view all the payslips for the batch.
```

ksc-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -2,173 +2,304 @@
+   :alt: Information that appears when generating payslips.
+
+.. note::
+   Payslips have a status of *Waiting* until the :guilabel:`Create Draft Entry` button is clicked.
+   After, the payslip status changes to *Done*.
```
In my runbot, and in your screenshot, I believe the statuses are: New, Confirmed, Done, and Paid. Can you confirm? If a change needs to be made, please adhere to the correct status names in the doc, if needed. Thanks 👍

larm-odoo replied:
I am going to remove this, because you see this in the Payslips - All Payslips part of Payroll, not in the Work Entries. It seems like too much info here, and does not reflect the image. I was just worried people would do this part, then wonder why they say 'waiting' in the payslips section if they did what they thought they needed to here.

ksc-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -2,173 +2,304 @@
-After conflicts have been resolved, work entries must be regenerated for each employee by clicking
-the :guilabel:`Regenerate Work Entries` button, and entering the corresponding information for each
-employee.
+To print payslips, first view the individual payslips by clicking the :guilabel:`Payslips` smart
+button on the batch form. Next, select the payslips to print in the payslip list. Click the box next
+to each payslip to print, or click the box next to :guilabel:`Reference` to select all the payslips
+in the list at once. Click the :guilabel:`Print` button, and a PDF file is created with all the
+specified payslips.
```
```suggestion
To print payslips, first view the individual payslips, by clicking the :guilabel:`Payslips` smart
button on the batch form.

Next, select the payslips to print from the :guilabel:`Payslips` list. Click the box next to each
payslip to print, or click the box to the left of the :guilabel:`Reference` column title, to select
all the payslips on the list at once.

Click the :guilabel:`Print` button, and a PDF file is created with all the specified payslips.
```

ksc-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -2,173 +2,304 @@
-Generating payslips
-===================
+.. note::
+   The :guilabel:`Print` button does not appear until at least one payslip is selected in the list.
```
```suggestion
   The :guilabel:`Print` button does **not** appear until at least one payslip is selected from the list.
```

ksc-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -2,173 +2,304 @@
+to next payslip`. This is because the employee was already paid for that day, and it was logged as
+time spent at work, as a typical work day.
+
+In order to keep the employee's time off balances correct, the time off request must be applied to
+the following pay period. This not only ensures time off request balances are current, but it also
+eliminates the need to redo work entries, cancel paychecks, and reissue paychecks.
```
```suggestion
In order to keep the employee's time off balances correct, the time off request **must** be applied
to the following pay period. This not only ensures time off request balances are current, it also
eliminates the need to redo work entries, cancel paychecks, and reissue paychecks.
```

ksc-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -2,173 +2,304 @@
+before the pay period ends, and an employee is unexpectedly sick on one of the last days of the pay
+period. The employee puts in a time off request for a day that was already processed on a payslip as
+a regular work day. Instead of canceling the payslip, modifying the work entries, and reissuing the
+paycheck, Odoo allows for those time off requests to be applied to the following pay period instead.
```
```suggestion
paycheck, Odoo allows for those time off requests to be applied to the following pay period, instead.
```

ksc-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -2,173 +2,304 @@
+:guilabel:`Payslip State` of :guilabel:`To defer to next payslip`.
+
+.. image:: work_entries/time-off-to-report.png
    :align: center
-   :alt: Generate Payslips button on the Work Entry dashboard.
+   :alt: A list of all time off requests that were not approved before payslips were generated.
```
it might be my comp, but this screenshot appears to be a *little* on the fuzzy side :) you may want to double-check and consider a fresh one during your revisions.

larm-odoo replied:
Agreed- that's what happens when I take too wide a shot! I cropped out the parts I didn't need and it's much more clear now.

ksc-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -2,173 +2,304 @@
-A batch entry will appear for the time period selected. The batch name appears at the top in the
-:guilabel:`Name` field, typically listing the month and year for the particular batch.
+To select the work entries to defer, click the box to the left of the work entry line. To select all
+work entries in the list, click the box next to :guilabel:`Employees` at the top of the list.
```
```suggestion
work entries in the list, click the box to the left of the :guilabel:`Employees` column title, at
the top of the list.
```

ksc-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -2,173 +2,304 @@
-.. note::
-   Payslips will have a status of *Waiting* until the :guilabel:`Create Draft Entry` button has been
-   clicked. After, the payslip status will change to *Done*.
+Click on an individual time off request and the details for that request load. The specific details
+for the time off request appear on the left-hand side, and all of the employee's submitted time off
+requests appear on the right-hand side (including the request in the details on the left-hand side).
```
```suggestion
Click on an individual time off request, and the details for that request load.

The specific details for the time off request appear on the left-hand side, and all of the
employee's submitted time off requests appear on the right-hand side (including the request in the
details on the left-hand side).
```

ksc-odoo reviewed (changes requested):
@larm-odoo another fantastically well-written, super-detailed, incredibly thorough doc. Way to go. Once you address the feedback I've left, and made the necessary changes, you can tag me again for another quick look. Keep up the great work! 👍

larm-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -2,173 +2,304 @@
+If the conflict is due to a time off request that has not been approved yet, a :guilabel:`Time Off`
+field appears on the left side, with the type of time off requested in the description.
+
+Time off conflicts
+------------------
+
+The most common work entry conflicts are for time off requests that have been submitted but not yet
+approved, which results in duplicate work entries for that employee (one for time off and another
+for regular work). If there is a conflict because a time off request is in the system for the same
+time that a regular work entry already exists, the time off request is entered in the
+:guilabel:`Time Off` field.
+
+Resolve on work entry
+~~~~~~~~~~~~~~~~~~~~~
+
+The time off conflict can be resolved either on the work entry pop-up, or on a detailed time off
+request pop-up. To resolve the time off conflict on this work entry pop-up, click the
+:guilabel:`Approve Time Off` button to approve the time off request and resolve the work entry
+conflict.
+
+The :guilabel:`Approve Time Off` and :guilabel:`Refuse Time Off` buttons disappear. Click the
+:guilabel:`Save & Close` button to close the pop-up. The conflict disappears from the work entry
+dashboard since the conflict is resolved.
```
I'll keep that in mind form now on!

larm-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -2,173 +2,304 @@
-   .. image:: work_entries/period.png
-      :align: center
-      :alt: Hours entered in the Period field.
+Conflicts are indicated with an orange triangle in the top-left corner of each individual work
+entry. Click on an individual work entry to see the date and time for the specific work entry, then
+click :guilabel:`Edit` to view the conflict details in a pop-up.
```
Added! Good idea.

larm-odoo commented:
Key @ksc-odoo - this is al set for you to take another look. I took all your edit advice-  I added the one you suggested, but then removed the one immediately in the section below since it was the same image, just with different buttons called out.  Just wanted you to know. Thanks again for your thorough review!

ksc-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -80,95 +91,245 @@ there are any errors on the work entry, such as required fields being left blank
-Click on an individual work entry to see the conflict details in a pop-up.
+The :guilabel:`Description`, :guilabel:`Employee`, and :guilabel:`Work Entry Type` are listed on
+the left side of the pop-up window. The :guilabel:`From` and :guilabel:`To` date and time range, as
+well as the total time (in hours), appears in the :guilabel:`Duration` field on the right side.
```
```suggestion
well as the total time (in hours) in the :guilabel:`Duration` field, appears on the right side.
```

ksc-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -80,95 +91,245 @@ there are any errors on the work entry, such as required fields being left blank
+Resolve on work entry
+~~~~~~~~~~~~~~~~~~~~~
+
+The time off conflict can be resolved either on the work entry pop-up window, or on a detailed time
+off request pop-up window.
```
I would move this portion *just* above the 'Resolve on work entry' section, as a sort of lead-in to the 2 sections that follow.

ksc-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -80,95 +91,245 @@ there are any errors on the work entry, such as required fields being left blank
+The time off conflict can be resolved either on the work entry pop-up window, or on a detailed time
+off request pop-up window.
+
+To resolve the time off conflict on this work entry pop-up, click the :guilabel:`Approve Time Off`
+button to approve the time off request, and resolve the work entry conflict.
```
```suggestion
To resolve the time off conflict on this work entry pop-up window, click the :guilabel:`Approve Time
Off` button to approve the time off request, and resolve the work entry conflict.
```

ksc-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -80,95 +91,245 @@ there are any errors on the work entry, such as required fields being left blank
-Link` button next to the :guilabel:`Time Off` entry will bring up the duplicate time-off request.
+To resolve the time off conflict on the detailed time off request pop-up window, click the
+:guilabel:`Internal Link` button at the end of the :guilabel:`Time Off` entry line, and the time off
+request details appear in a pop-up window. The request can be modified, if needed.
```
```suggestion
request details appear in a new pop-up window. The request can be modified, if needed.
```

ksc-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -80,95 +91,245 @@ there are any errors on the work entry, such as required fields being left blank
-.. image:: work_entries/batch.png
-   :align: center
-   :alt: Batch information that appears when making a batch.
+The date range to which the payslips apply to appears in the :guilabel:`Period` field, and the
+company appears in the :guilabel:`Company` field. It is **not** possible to make changes to this
+form.
```
```suggestion
The date range to which the payslips apply appears in the :guilabel:`Period` field, and the
company appears in the :guilabel:`Company` field. It is **not** possible to make changes to this
form.
```

ksc-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -80,95 +91,245 @@ there are any errors on the work entry, such as required fields being left blank
+==================
+
+If a time off request is submitted for a time period that was already processed on a payslip, the
+time off request appears in the :guilabel:`Time Off to Report`, with a status of :guilabel:`To defer
+to next payslip`. This is because the employee was already paid for that day, and it was logged as
+time spent at work, as a typical work day.
```
```suggestion
time off request appears in the :guilabel:`Time Off` page in the *Payroll* app, which is accessible
by navigating to :menuselection:`Payroll app --> Work Entries --> Time Off to Report`. 

On the :guilabel:`Time Off` page, the request appears with a status of :guilabel:`To defer to next
payslip`. This is because the employee was already paid for that day, and it was logged as time
spent at work, as a typical work day.
```

ksc-odoo reviewed (approved):
@larm-odoo Great work on making all the revisions on this PR! I only have a small handful of adjustments that need your attention. But I'm going to go ahead and approve this now, because I don't think you'll have any issues making the necessary changes during your next round of revision. Once you got this one good to go, feel free to tag it for Tech Review. 👍

larm-odoo commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -2,75 +2,86 @@
+If any entries have :ref:`conflicts <payroll/conflicts>` that need to be resolved, the dashboard
+defaults to filter only the `Conflicting` entries.
+
+To remove the filter from the :guilabel:`Search...` bar to view all work entries, click the
```
OK no problem! But its in a lot of my docs, so I hope I catch them all moving forward =D

larm-odoo commented:
Thanks again, @ksc-odoo! I appreciate it, as always. This is now ready for your tech review, @samueljlieber.

samueljlieber commented on content/applications/hr/payroll/work_entries/generate-payslips.png:
Please compress this image `generate-payslips.png`

samueljlieber commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -2,75 +2,86 @@
+first name of the employees. The entire month is displayed, with the current day highlighted in pale
+yellow.
+
+If any entries have :ref:`conflicts <payroll/conflicts>` that need to be resolved, the dashboard
+defaults to filter only the `Conflicting` entries.
```
:guilabel:\`Conflicting\` was used later in the doc, and I like using the :guilabel: when referring to the active filters. Can use literal formatting when typing in to apply a filter.
```suggestion
If any entries have :ref:`conflicts <payroll/conflicts>` that need to be resolved, the dashboard
defaults to filter only the :guilabel:`Conflicting` entries.
```

samueljlieber commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -2,75 +2,86 @@
+If any entries have :ref:`conflicts <payroll/conflicts>` that need to be resolved, the dashboard
+defaults to filter only the `Conflicting` entries.
+
+To remove the filter from the :guilabel:`Search` bar to view all work entries, click the
+:guilabel:`✖️ (remove)` icon on the `Conflicting` filter in the :guilabel:`Search` bar, and all
+work entries appear in the list.
```
Please match the text on the UI exactly when using :guilabels: (:guilabel:\`Search\` ➡️ :guilabel:\`Search...\`)
```suggestion
To remove the filter from the search bar to view all work entries, click the :guilabel:`✖️ (remove)`
icon on the :guilabel:`Conflicting` filter in the :guilabel:`Search...` bar, and all work entries
appear in the list.
```
or
```suggestion
To remove the filter from the :guilabel:`Search...` bar to view all work entries, click the
:guilabel:`✖️ (remove)` icon on the :guilabel:`Conflicting` filter in the :guilabel:`Search...` bar,
and all work entries appear in the list.
```

larm-odoo replied:
Ahhh, ok. When KSC did my last review, he changed it to the GUI 'Search...' like you have in the second option, but then changed it in a 2nd review to remove the "...".  I do agree with you , that is how it appears, so that's how it should look. I'll see if we can add this to our linter/style guide so we can be consistent.

samueljlieber commented on content/applications/hr/payroll/work_entries.rst:
```diff
@@ -80,95 +91,247 @@ there are any errors on the work entry, such as required fields being left blank
+==================
+
+If a time off request is submitted for a time period that was already processed on a payslip, the
+time off request appears in the :guilabel:`Time Off` page in the *Payroll* app, which is accessible
+by navigating to :menuselection:`Payroll app --> Work Entries --> Time Off to Report`.
```
No need for a guilabel on the "_Time Off_ page" since that text is not on the active UI
```suggestion
time off request appears in the *Time Off* page in the *Payroll* app, which is accessible by
navigating to :menuselection:`Payroll app --> Work Entries --> Time Off to Report`.
```

samueljlieber reviewed (changes requested):
Hi @larm-odoo, nice work on the improvements in this PR. Overall everything is looking great, I just have a few technical corrections, please see below.

Tag me for another quick look once these have been addressed, thank you!

larm-odoo commented:
All set for you, @samueljlieber!

samueljlieber reviewed (changes requested):
Hi @larm-odoo, last thing– there are only 11 `.. image::` directives in the `work_entries.rst` doc, but there are 20 images in the `work_entries/` media folder. Please remove each of the unused images 🙂

| USED:                     | UNUSED:                 |
|---------------------------|-------------------------|
| work-entries-overview.png | batch.png               |
| create.png                | conflict-detail.png     |
| conflict-pop-up.png       | conflict-duplicate.png  |
| conflict-details.png      | conflicts.png           |
| time-off-details.png      | create-payslips.png     |
| regenerate-details.png    | payslips.png            |
| generate-payslips.png     | period.png              |
| print-payslips.png        | regenerate-employee.png |
| time-off-to-report.png    | validate.png            |
| batch-defer.png           |                         |
| single-defer.png          |                         |

larm-odoo commented:
@samueljlieber I didn't realize I had unused images- they have all been removed, so only the ones listed in the doc are in the folder now. All set!

samueljlieber reviewed (approved):
Looks good to me, thank you @larm-odoo!

@StraubCreative this PR is good to go 👍

StraubCreative reviewed (approved):
@robodoo r+
